docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
-## Design & implementation
+- [AB3](https://github.com/se-edu/addressbook-level3)
+- [OpenCSV](https://opencsv.sourceforge.net/)
+- [gson](https://github.com/google/gson)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Introduction 
 
+Welcome to rainyDay's developer guide! rainyDay is a desktop application for managing your finances that runs on a Command Line Interface (CLI). rainyDay
+provides a simple solution to track your finances and achieve your financial goals. 
+
+#### Main functions:
+* Adding, deleting and editing transaction entries
+* Viewing your transactions and filtering by certain fields
+* Saving your data into a csv file
+* Setting budgeting goals
+
+#### Purpose and target reader
+
+This guide is designed for potential engineers  and professors who may want to who may want to understand how rainyDay is implemented. 
+We assume that you already have a certain level of technical knowledge as we will be going into detail of the technical aspects
+of rainyDay. <br> If you are a user looking for instructions on how to use rainyDay, please refer to our [UserGuide](https://ay2223s2-cs2113t-t09-1.github.io/tp/UserGuide.html)
+
+#### How to use this guide
+
+{todo, talk about the diagrams, formatting (markdown etc)}
+
+## Design
+
+> 💡**Tip:** The .puml files used to create diagrams in this document can be found [here](https://github.com/AY2223S2-CS2113T-T09-1/tp/tree/master/docs/diagrams). 
+> Refer to the PlantUML Tutorial at se-edu/guides to learn how to create and edit diagrams.
+
+### Architecture
+
+#### Main components of the architecture
```

Comment:
Might be better for these to reflect the actual classes used in the code:
- `Main` -> `RainyDay`
- `Userdata` -> `UserData`
- `UI` -> `Ui`